src/modules/dashboard/components/ColonyActions/ColonyActions.tsx

```diff
@@ -60,118 +62,162 @@ type Props = {
+          return 0;
+      }
+    },
+    [actionsSortOption],
+  );
+
+  const sortedActionsData: FormattedAction[] = useMemo(
+    () => actions.sort(actionsSort),
+    [actionsSort, actions],
+  );
+
+  const handleActionRedirect = useCallback(
+    ({ transactionHash }: RedirectHandlerProps) =>
+      history.push(`/colony/${colonyName}/tx/${transactionHash}`),
+    [colonyName, history],
+  );
+
+  if (
+    paymentActionsLoading ||
+    commentCountLoading ||
+    !paymentActions ||
+    !commentCount
+  ) {
+    return (
+      <div className={styles.loadingSpinner}>
+        <SpinnerLoader
+          loadingText={MSG.loading}
+          appearance={{ theme: 'primary', size: 'massive' }}
+        />
+      </div>
+    );
+  }
 
   return (
     <div className={styles.main}>
-      <Form
-        initialValues={{ filter: ActionFilterOptions.ENDING_SOONEST }}
-        onSubmit={() => undefined}
-      >
-        <div className={styles.filter}>
-          <Select
-            appearance={{ alignOptions: 'left', theme: 'alt' }}
-            elementOnly
-            label={MSG.labelFilter}
-            name="filter"
-            options={ActionFilterSelectOptions}
-            onChange={setActionsFilter}
-            placeholder={MSG.placeholderFilter}
+      {sortedActionsData?.length ? (
+        <>
+          <Form
+            initialValues={{ filter: ActionsSortOptions.NEWEST }}
+            onSubmit={() => undefined}
+          >
+            <div className={styles.filter}>
+              <Select
+                appearance={{ alignOptions: 'left', theme: 'alt' }}
+                elementOnly
+                label={MSG.labelFilter}
+                name="filter"
+                options={ActionsSortSelectOptions}
+                onChange={setActionsSortOption}
+                placeholder={MSG.placeholderFilter}
+              />
+            </div>
+          </Form>
+          <ActionsList
+            items={sortedActionsData}
+            handleItemClick={handleActionRedirect}
+            colony={colony}
           />
-        </div>
-      </Form>
-      <ActionsList
-        items={filteredActionsData.slice(0, mockDataPager * ITEMS_PER_PAGE)}
-        handleItemClick={handleActionRedirect}
-      />
-      {mockDataPager * ITEMS_PER_PAGE < MOCK_ACTIONS.length && (
-        <div className={styles.controls}>
-          <Button
-            appearance={{ size: 'medium', theme: 'primary' }}
-            onClick={fakeFetchMoreData}
-            text={MSG.loadMore}
-            loading={fakeFetchingData}
+        </>
+      ) : (
+        <div className={styles.emptyState}>
+          <FormattedMessage
+            {...MSG.noActionsFound}
+            values={{
+              /*
+               * @TODO Maybe make this check smarter?
+               *
+               * By injecting a env var when starting dev:heavy and actually
+               * knowing if the subgraph process was started, rather then
+               * just guessing...
```

Comment:
Not `isDevMode`, more something like `isSubgraphRunning` than we can differentiate between `isDevMode` and `isSubgraph`, because there may be instances where you run dev "light"